### PR TITLE
Duplicate user fields when duplicate field groups exists. #2615

### PR DESCRIPTION
**Pull request by MaximilianoRicoTabo**
* Bail if the group already exists in pmpro_add_field_group function

### All Submissions:

* [x] Have you followed the [Contributing guideline](https://github.com/strangerstudios/paid-memberships-pro/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/paid-memberships-pro/pulls) for the same update/change?

### Changes proposed in this Pull Request:

 Check if group name exist before add it. From ou UI this won't happen because it handles in a way that adds a number within parenthesis to the name if it's repeated before submit the form in a jQuery event handler.  So, despite it's an edge case probably causes some trouble to support team and it's good to handle.

Resolves  https://github.com/strangerstudios/paid-memberships-pro/issues/2615

### How to test the changes in this Pull Request:

1. Follow steps from issue above
2.  Observe if you see duplicates or not

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

Added validation for duplicated names in pmpro_add_field_group function

### CODE REVIEW

File: includes/fields.php [hunk 1]
```diff
@@ -78,6 +78,12 @@ function pmpro_add_user_field( $where, $field ) {
  */
 function pmpro_add_field_group( $name, $label = NULL, $description = '', $order = NULL ) {
 	global $pmpro_field_groups;
+	//Bail if the group already exists.
+	foreach($pmpro_field_groups as $group) {
+		if($group->name == $name)
+		//already exists	
+		return false;
+	}
```
Comment (ideadude) [inline includes/fields.php:86]:
```suggestion
	//Bail if the group already exists.
	foreach( $pmpro_field_groups as $group ) {
		if( $group->name === $name ) {
			//already exists
			return false;
		}
	}
```
WPCS and using === here. Both values should always be strings, but just in case.

**Review by ideadude — COMMENTED** [review]
Minor WPCS fix.

I'm also wondering if we should return false here or maybe overwrite the group info with the new data on the second call?

Do we know what the user would intend?

When is this coming up most often? I assume maybe there is a group in the user field settings in the admin, then we run some custom code afterward that is trying to add the groups. In this case, should the code (or whatever call comes later) overwrite the group label and description then?

File: includes/fields.php (hunk 1 above)
Comment (MaximilianoRicoTabo) [reply includes/fields.php]: Thanks. Added the suggested change. Regarding  the questions 

> Do we know what the user would intend?
> When is this coming up most often?

From the issue itself it seems it comes from custom code added to functions.php file in the theme or snippet plugin. Actually, we deal with dups in our UI by adding a number between parenthesis. We may try that approach for dups coming from custom code as well, but also we may enforce users to adopt user fields from UI, i guess is something that comes from RH deprecation and should be an edge case and should tend to stop happening.

File: includes/fields.php [hunk 2]
```diff
@@ -78,6 +78,12 @@ function pmpro_add_user_field( $where, $field ) {
  */
 function pmpro_add_field_group( $name, $label = NULL, $description = '', $order = NULL ) {
 	global $pmpro_field_groups;
+	//Bail if the group already exists.
+	foreach($pmpro_field_groups as $group) {
+		if($group->name === $name)
+		//already exists	
+		return false;
+	}
```
Comment (dparker1005) [inline includes/fields.php:86]: WPCS
```suggestion
	// Bail if the group already exists.
	foreach ( $pmpro_field_groups as $group ) {
		if ( $group->name === $name ) {
			// Group already exists.
			return false;
		}
	}
```
